### DISCUSSION TITLE
[Question] How to use arguments in on_click

### DISCUSSION THREAD

**Discussion by Dzheremi2** [flet-dev/flet#3970] [Q&A] [answered]
### Question

I'm trying to call on_click function with args like this 
`on_click=gen_line_inputs(some args)`
but it isn't working.
The I tried to use this `on_click=gen_line_inputs, args=(page, 12,))` method I found on streamlit, but it also didn't working.

So how can I call that func with args

### Code sample

```python
# Button with on_click

proceedLyrcis = ft.FloatingActionButton(
        icon=ft.icons.CHECK_CIRCLE_OUTLINE_OUTLINED,
        text="Start sync",
        enable_feedback=True,
            on_click=gen_line_inputs
    )

# The Function

def gen_line_inputs(page, lines_cnt):
    sync_fields_column = ft.Column()
    page.controls.append(sync_fields_column)
    for i in range(lines_cnt):
        sync_fields_column.controls.append(editLine)
    page.update()
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by vishnu-rvb** [comment, marked as answer]
you can use lambda function.
`on_click=lambda event:gen_line_inputs(page, lines_cnt)` assuming page and lines_cnt will be kept static.
in case of dynamic arguments , assign values to your arguments and set them as defaults.
`on_click=lambda event,page=page,lines_cnt=lines_cnt:gen_line_inputs(page, lines_cnt)`

**Reply by Dzheremi2** [reply]
> you can use lambda function. `on_click=lambda event:gen_line_inputs(page, lines_cnt)` assuming page and lines_cnt will be kept static. in case of dynamic arguments , assign values to your arguments and set them as defaults. `on_click=lambda event,page=page,lines_cnt=lines_cnt:gen_line_inputs(page, lines_cnt)`

Thank you for help, but now I have another problem. After adding all necessary content, I need to update `page` with `page.update()` from my function, **which is in another file**. But when I try to do so, I get this error:

`TypeError: Control._build_add_commands() missing 1 required positional argument: 'self'`

Can you help me if it's not too much trouble.

**Reply by vishnu-rvb** [reply]
are you implementing flet controls from inside a class or from a simple main function?
show the full error message and form which line of code

**Reply by Dzheremi2** [reply]
Thanks for your answer a lot❤️ but i just forgot to place brackets in object call inside my function.

I used
```python
sync_fields_column_controls.append(editLine)
```
insead of 
```python
sync_fields_column_controls.append(editLine())
```

And this was an error. Sorry for an inconvenience.
